client/src/templates/Introduction/components/CertChallenge.js

```diff
@@ -53,10 +69,14 @@ const CertChallenge = ({
   t,
   verifyCert,
   title,
+  fetchState,
+  isSignedIn,
   user: { isHonest, username }
 }) => {
   const [canClaim, setCanClaim] = useState({ status: false, result: '' });
   const [isCertified, setIsCertified] = useState(false);
+  const [userLoaded, setUserLoaded] = useState(false);
+  const [userChecked, setUserChecked] = useState(false);
```

Comment:
What does `userChecked` mean?  I think it would make more sense to have a `verificationFetchState` object to track `getVerifyCanClaimCert`.  Also, if you're feeling keen, `canClaim` really should be a boolean called `certIsClaimable` and `result` should be separate and called something meaningful. `verificationStatus`, maybe.  It's unclear to me what it's for.

@ShaunSHamilton might have more context.

Reply:
Not sure what you mean by :

>  I think it would make more sense to have a verificationFetchState object to track getVerifyCanClaimCert

Reply:
So, we generally use an object

```
{   
  pending: boolean,
  complete: boolean,
  errored: boolean,
  error: null | string
}
```
to track the status of an async operation, so I was suggesting we do something similar here.  Essentially we perform an ajax operation and we conditionally display stuff on the page depending on what state that operation is in.

On reflection, it's probably over the top to use the full object if we only care about the `complete` part of the process.

Anyway, that's a lot of words and not a very concrete recommendation!  So, I'll suggest a code change in another comment.

Reply:
So, I figured it was easier to show than tell, and I pushed a commit.

The rationale for the name changes is pretty straightforward.  I don't know what `canClaim.status` or `canClaim.result` mean, but I can easily guess about `canClaimCert` and `certVerificationMessage`.  Well, I still don't know what the latter means, but it *is* a message about certification verification :man_shrugging: 

Same point for `userChecked` vs `verificationComplete`.  The former is a bit vague (what are we checking the user _for_). The latter is quite explicit and mirrors the patterns we usually use for ajax requests.  (pending, complete, etc. etc.).